app/src/main/java/com/stepango/archetype/player/loader/DownloadEpisodeAction.kt

```diff
@@ -0,0 +1,51 @@
+package com.stepango.archetype.player.loader
+
+import android.content.Context
+import com.stepango.archetype.action.*
+import com.stepango.archetype.logger.logger
+import com.stepango.archetype.player.data.db.EpisodesModelRepo
+import com.stepango.archetype.player.data.db.model.EpisodesModel
+import com.stepango.archetype.player.di.Injector
+import com.stepango.archetype.player.di.lazyInject
+import com.stepango.archetype.util.getFileName
+import io.reactivex.Completable
+import io.reactivex.schedulers.Schedulers.io
+import java.io.File
+
+/**
+ * Wild, 02.07.2017.
+ */
+
+class RefreshDownloadedFilesAction : ContextAction {
+
+    val episodesRepo by lazyInject { episodesRepo() }
+
+    override fun invoke(context: Context, args: Args): Completable
+            = episodesRepo.getAll()
+            .subscribeOn(io())
+            .flatMapCompletable {
+                val file = File(context.filesDir, getFileName(it.audioUrl))
+                if (file.exists()) {
+                    logger.d("file exists for ${it.name}")
+                    it.file = file.absolutePath
```

Comment:
episode model should be immutable.
make EpisodeModel be data class and then you can call it.copy{file = file.absolutePath}